snudown.h

```diff
@@ -0,0 +1,118 @@
+/* snudown.h - header for snudown */
```

Comment:
It's not really a maintenance problem, it's just generally considered bad form to have the function implementation itself in the `.h` instead of the `.c`. Having the function _definition_ in the `.h` is OK and generally what you want. 

Any function implementation longer than 1-2 lines should be in a `.c` it only needs to be compiled to a single `.o` and the implementation won't need to be recompiled for every object with a `.c` file with `#include "snudown.h"`.

Also, if you put the implementation in the `.h` rather than the `.c`, any change to the function implementation will trigger a recompile of any objects that reference `snudown.h`. If the implementation is somewhere like `renderers.c`, then a change to the implementation will only require a recompile of that specific object file and a re-link.